### DISCUSSION TITLE
There is App.bind but no App.unbind

### DISCUSSION THREAD

**Discussion by arcivanov** [Textualize/textual#4644] [Q&A] [answered]
I'm trying to have dynamic bindings based on current tab in TabbedContent. While there are ways to process existing global bindings taking into account the current tab, and there is a way to dynamically add binding there doesn't seem to be a way to unbind existing bindings dynamically.
What's the framework-standard way for context-based switching of active bindings?

Thanks!

**Comment by davep** [comment, marked as answer]
If you want specific bindings to only be available in a certain tab pane, or even the *same* key but bound in different ways in different tab panes, the usual way would be to make those bindings part of the pane. For example, here's 3 different tabs in a `TabbedContent`, each one has a binding for <kbd>F1</kbd>, but each one does different things:

```python
from textual.app import App, ComposeResult
from textual.binding import Binding
from textual.widgets import Footer, TabbedContent, TabPane, OptionList

class FooTab(TabPane):

    BINDINGS = [
        Binding("f1", "gndn", "Does a Foo"),
    ]

    def compose(self) -> ComposeResult:
        yield OptionList(*[f"Foo {n}" for n in range(100)])


class BarTab(TabPane):

    BINDINGS = [
        Binding("f1", "gndn", "Does a Bar"),
    ]

    def compose(self) -> ComposeResult:
        yield OptionList(*[f"Bar {n}" for n in range(100)])

class BazTab(TabPane):

    BINDINGS = [
        Binding("f1", "gndn", "Does a Baz"),
    ]

    def compose(self) -> ComposeResult:
        yield OptionList(*[f"Baz {n}" for n in range(100)])

class TabbedBindingsApp(App[None]):

    CSS = """
    OptionList {
        height: 1fr;
    }
    """

    def compose(self) -> ComposeResult:
        with TabbedContent():
            yield FooTab("Foo")
            yield BarTab("Bar")
            yield BazTab("Baz")
        yield Footer()

if __name__ == "__main__":
    TabbedBindingsApp().run()
```

**Reply by arcivanov** [reply]
Let me try this approach but with on_mount. Thank you!

**Reply by arcivanov** [reply]
This also works (although a non-standard way):

```
def on_mount(self):
        bindings = self.query_one("#tab-orders", TabPane)._bindings
        bindings.bind("o", "new_order", "New Order")
```

**Reply by arcivanov** [reply]
Actually, while the bindings dynamically appear in the footer, they don't appear to be working (triggering) with my `on_mount` approach.

**Reply by arcivanov** [reply]
Ah, that's because the actions are in the app and not in the TabPane itself.

**Reply by davep** [reply]
I'd also strongly caution against making use of internals like that.

**Reply by arcivanov** [reply]
> I'd also strongly caution against making use of internals like that.

Naturally.
